gui/test/integration/cypress/support/page_files/NodeSearchInput.ts

```diff
@@ -0,0 +1,102 @@
+class NodeSearchSuggestionLink {
+  constructor(
+    private readonly index: number,
+    private readonly searchSuggestions: NodeSearchSuggestions
+  ) {}
+
+  get() {
+    return this.searchSuggestions.get().find("ul>a").eq(this.index);
+  }
+
+  click() {
+    this.get().click();
+
+    return this;
+  }
+}
+
+class NodeSearchAllResultsLink {
+  constructor(private readonly searchSuggestions: NodeSearchSuggestions) {}
+
+  get() {
+    return this.searchSuggestions.get().find("ul>a:last-child");
+  }
+
+  click() {
+    this.get().click();
+
+    return this;
+  }
+}
+
+export class NodeSearchSuggestions {
+  // The component `div` is actually created in the root component
+  // so it has 'no parent elements'
+  readonly selector: string = "[data-cy=nodeSearchSuggestions]";
+
+  get() {
+    return cy.get(this.selector);
+  }
+
+  assertNoResults() {
+    this.get().find("ul").children().should("have.length", 1);
+    this.get().find("ul>li").should("have.text", "No Results");
+
+    return this;
+  }
+
+  suggestion(index: number) {
+    return new NodeSearchSuggestionLink(index, this);
+  }
+
+  clickSuggestion(index: number) {
+    this.suggestion(index).click();
+
+    return this;
+  }
+
+  get allResults() {
+    return new NodeSearchAllResultsLink(this);
+  }
+
+  clickAllResults() {
```

Comment:
More redundant code. I like the style of getters returning sub-[page-]objects that can then be manipulated at a high level (click, get text, whatever). I don't think we also need a completely fluent interface, especially since the two goals are at odds with each other.

Reply:
Makes sense, removed all the clicks and kept the `enter` for search input